### PR TITLE
Adding support for protobuf decimal

### PR DESCRIPTION
**Pull request by allkliu**
In this PR, we define a standard decimal.proto in the library, as Protobuf's standard library and WellKnownTypes currently do not provide a Decimal type. The existing types cannot accurately represent some decimals, so we would like to add this proto to represent such decimals for customers using Glue Schema Registry.

Related Issue: https://github.com/protocolbuffers/protobuf/issues/4406

decimal.proto is defined below (with comments removed for clarity):

    syntax = "proto3";

    package decimal;

    option java_package = "decimal";
    option java_outer_classname = "ProtobufSchemaDecimal";

    message Decimal {
      int64 units = 1;
      int32 fraction = 2;
      uint32 precision = 3;
      int32 scale = 4;
    }

### CODE REVIEW

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
 import com.squareup.wire.schema.internal.parser.TypeElement;
 import kotlin.ranges.IntRange;
 import metadata.ProtobufSchemaMetadata;
+import decimal.ProtobufSchemaDecimal;
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:51]: This package can be named `additionalTypes.Decimal`

File: utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java [hunk 2]
```diff
@@ -383,7 +385,7 @@ private static boolean isParentLevelType(ProtoType protoType, Optional<String> o
 
         //In case the package is not defined, we select the types that are not google types or metadata types.
         return !typeName.startsWith("google.type") && !typeName.startsWith("google.protobuf")
-            && !typeName.startsWith("metadata");
+            && !typeName.startsWith("metadata") && !typeName.startsWith("decimal");
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java:388]: Minor: New line.

File: utils/protobuf-schema-utilities/src/main/proto/decimal/decimal.proto [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+package decimal;
+
+option java_package = "decimal";
+option java_outer_classname = "ProtobufSchemaDecimal";
+
+// Protobuf standard library / WellKnownTypes don't provide a Decimal type.
+// This posses a challenge to represent Decimals that cannot be accurately represented
+// using float or double types. This Proto provides a way to represent Decimals for
+// customers using Glue Schema Registry.
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/main/proto/decimal/decimal.proto:11]: Glue Schema Registry is not relevant here as this is general purpose Apicurio library. Please remove all other references.

File: utils/protobuf-schema-utilities/src/test/proto/decimal/decimal.proto [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+option java_package = "decimal";
+option java_outer_classname = "ProtobufSchemaDecimal";
+
+// Protobuf standard library / WellKnownTypes don't provide a Decimal type.
+// This posses a challenge to represent Decimals that cannot be accurately represented
+// using float or double types. This Proto provides a way to represent Decimals for
+// customers using Glue Schema Registry.
+// Related issue: https://github.com/protocolbuffers/protobuf/issues/4406
+message Decimal {
```
Comment (blacktooth) [inline utils/protobuf-schema-utilities/src/test/proto/decimal/decimal.proto:13]: Do we need to duplicate this? Can the one in sources be used? Also, please remove references to Glue Schema Registry.

Comment (allkliu) [reply utils/protobuf-schema-utilities/src/test/proto/decimal/decimal.proto:13]: I'm not sure - Kexin has both in her metadata changes? I will try with just one

**Comment by blacktooth** [comment]
Looks good! Approved

**Comment by hhkkxxx133** [comment]
@carlesarnal Can you please prioritize this? Also may I ask when the next release of the library is planned? Thanks in advance!

**Comment by carlesarnal** [comment]
> @carlesarnal Can you please prioritize this? Also may I ask when the next release of the library is planned? Thanks in advance!

There's no ETA for an upstream release. That said, upstream releases are cheap and we have some work ready to be released, so expect it to be soon.

cc @EricWittmann

**Comment by hhkkxxx133** [comment]
Thanks for the reply! @EricWittmann Any plan or ETA we can track for releasing the library? We are depending on the release to continue our development work. Thanks in advance!

**Pull request merged**